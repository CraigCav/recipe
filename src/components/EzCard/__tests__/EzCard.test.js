import React from 'react';
import {EzCard, EzCardSection} from '../';

describe('EzCard', () => {
  it('should render with default styles', () => {
    const actual = create(
      <EzCard>
        <p>Lorem ipsum dolor</p>
      </EzCard>
    );
    expect(actual).toMatchSnapshot();
  });

  it('should render with header styles', () => {
    const actual = create(
      <EzCard title="Card Heading">
        <p>Lorem ipsum dolor</p>
      </EzCard>
    );
    expect(actual).toMatchSnapshot();
  });

  it('should render with header and subheader styles', () => {
    const actual = create(
      <EzCard title="Card Heading" subtitle="Card Subheading">
        <p>Lorem ipsum dolor</p>
      </EzCard>
    );
    expect(actual).toMatchSnapshot();
  });

  it('should render with card sections', () => {
    const actual = create(
      <EzCard>
        <EzCardSection>Lorem ipsum dolor</EzCardSection>
        <EzCardSection>Lorem ipsum dolor</EzCardSection>
      </EzCard>
    );
    expect(actual).toMatchSnapshot();
  });

  it('should render card sections horizontally', () => {
    const actual = create(
      <EzCard horizontal>
        <EzCardSection>Lorem ipsum dolor</EzCardSection>
        <EzCardSection>Lorem ipsum dolor</EzCardSection>
      </EzCard>
    );
    expect(actual).toMatchSnapshot();
  });

  it('should meet accessibility guidelines', async () => {
    const wrapper = renderToHtml(
      <EzCard title="Card Heading">
        <EzCardSection>Lorem ipsum dolor</EzCardSection>
        <EzCardSection>Lorem ipsum dolor</EzCardSection>
      </EzCard>
    );
    const actual = await axe(wrapper);
    expect(actual).toHaveNoViolations();
  });

  describe('valid props', () => {
    let actual;
    beforeEach(() => {
      actual = shallow(
        <EzCard>
          <EzCardSection>Lorem ipsum dolor</EzCardSection>
          <EzCardSection>Lorem ipsum dolor</EzCardSection>
        </EzCard>
      );
    });
    it('renders valid props for html elements', () => {
      actual.setProps({'data-test': 'my-test-selector'});
      expect(actual.prop('data-test')).toEqual('my-test-selector');
    });

    it('does NOT render invalid props for html elements', () => {
      expect(actual.prop('primary')).toBeUndefined();
    });
  });
});
