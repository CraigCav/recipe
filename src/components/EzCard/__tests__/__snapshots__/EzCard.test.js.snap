// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EzCard should render card sections horizontally 1`] = `
.recipe-6 {
  background: #FFFFFF;
  border: 1px solid #CED4D9;
  border-radius: 6px;
}

.recipe-0 {
  padding: 1.4285714285714286rem;
}

.recipe-0 > *:not(:first-child) {
  margin-top: 1.1428571428571428rem;
}

.recipe-4 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.recipe-4 > * {
  -webkit-flex-basis: 0;
  -ms-flex-preferred-size: 0;
  flex-basis: 0;
  -webkit-box-flex: 1;
  -webkit-flex-grow: 1;
  -ms-flex-positive: 1;
  flex-grow: 1;
}

<div
  className="recipe-6 recipe-7"
>
  <div
    className="recipe-4 recipe-5"
  >
    <div
      className="recipe-0 recipe-1"
    >
      Lorem ipsum dolor
    </div>
    <div
      className="recipe-0 recipe-1"
    >
      Lorem ipsum dolor
    </div>
  </div>
</div>
`;

exports[`EzCard should render with card sections 1`] = `
.recipe-6 {
  background: #FFFFFF;
  border: 1px solid #CED4D9;
  border-radius: 6px;
}

.recipe-4 > *:not(:first-child) {
  border-top: 1px solid #CED4D9;
}

.recipe-0 {
  padding: 1.4285714285714286rem;
}

.recipe-0 > *:not(:first-child) {
  margin-top: 1.1428571428571428rem;
}

<div
  className="recipe-6 recipe-7"
>
  <div
    className="recipe-4 recipe-5"
  >
    <div
      className="recipe-0 recipe-1"
    >
      Lorem ipsum dolor
    </div>
    <div
      className="recipe-0 recipe-1"
    >
      Lorem ipsum dolor
    </div>
  </div>
</div>
`;

exports[`EzCard should render with default styles 1`] = `
.recipe-4 {
  background: #FFFFFF;
  border: 1px solid #CED4D9;
  border-radius: 6px;
}

.recipe-2 > *:not(:first-child) {
  border-top: 1px solid #CED4D9;
}

.recipe-0 {
  padding: 1.4285714285714286rem;
}

.recipe-0 > *:not(:first-child) {
  margin-top: 1.1428571428571428rem;
}

<div
  className="recipe-4 recipe-5"
>
  <div
    className="recipe-2 recipe-3"
  >
    <div
      className="recipe-0 recipe-1"
    >
      <p>
        Lorem ipsum dolor
      </p>
    </div>
  </div>
</div>
`;

exports[`EzCard should render with header styles 1`] = `
.recipe-6 {
  background: #FFFFFF;
  border: 1px solid #CED4D9;
  border-radius: 6px;
}

.recipe-4 > *:not(:first-child) {
  border-top: 1px solid #CED4D9;
}

.recipe-2 {
  padding: 1.4285714285714286rem;
}

.recipe-2 > *:not(:first-child) {
  margin-top: 1.1428571428571428rem;
}

.recipe-0 {
  font-size: 1.25rem;
  margin: 1.4285714285714286rem 1.4285714285714286rem 0;
}

<div
  className="recipe-6 recipe-7"
>
  <h3
    className="recipe-0 recipe-1"
  >
    Card Heading
  </h3>
  <div
    className="recipe-4 recipe-5"
  >
    <div
      className="recipe-2 recipe-3"
    >
      <p>
        Lorem ipsum dolor
      </p>
    </div>
  </div>
</div>
`;
